Contributed by Erwin Andreasen:

The modification I'd suggest is to make unzip unzip only
s3m/mod/mtm/ult/xm files from the zip archives.

--- load_mod.c.orig	Sat Feb  1 15:25:15 1997
+++ load_mod.c	Sat Feb  1 20:06:27 1997
@@ -569,8 +569,8 @@ load_module (char *name, struct song_inf
 	  break;
 	case ZIPCOMPRESSED:
 	  command = (char *) malloc (strlen (fixedname) +
-				     strlen (ZIPDECOMP_PGM) + 1);
-	  sprintf (command, "%s%s", ZIPDECOMP_PGM, fixedname);
+				     strlen (ZIPDECOMP_PGM) + strlen(ZIPDECOMP_PARAMS) + 1);
+	  sprintf (command, "%s%s%s", ZIPDECOMP_PGM, fixedname, ZIPDECOMP_PARAMS);
 	  break;
 	default:
 	  break;
--- defines.h.orig	Sat Feb  1 14:30:52 1997
+++ defines.h	Sat Feb  1 16:01:42 1997
@@ -141,7 +141,8 @@
 /* modified and added to by Peter Federighi */
 #define GDECOMP_PGM	"gunzip -c "	/* program for decompression */
 #define LHADECOMP_PGM	"lha pq "	/* lha decompression */
-#define ZIPDECOMP_PGM	"unzip -p "	/* zip decompression */
+#define ZIPDECOMP_PGM	"unzip -qq -C -p "	/* zip decompression: Quiet, Case Insensitive, Pipe */
+#define ZIPDECOMP_PARAMS " *s3m *mod *ult *669 *xm *mtm 2> /dev/null"
 
 #define GCOMPRESSED	2	/* start with 2 since FALSE */
 #define LHACOMPRESSED	3	/* and TRUE take up 0 and 1 */
